jobs/webcompat-kb/webcompat_kb/main.py

```diff
@@ -1024,6 +1024,73 @@ def fetch_update_history(self, all_bugs: BugsById) -> list[BugHistoryEntry]:
 
         return []
 
+    def update_metric_history(self) -> None:
+        metrics_table = f"{self.bq_dataset_id}.webcompat_topline_metric"
+        history_table = f"{self.bq_dataset_id}.webcompat_topline_metric_history"
+
+        history_schema = [
+            bigquery.SchemaField("recorded_date", "DATE", mode="REQUIRED"),
+            bigquery.SchemaField("date", "DATE", mode="REQUIRED"),
+            bigquery.SchemaField("bug_count", "INTEGER", mode="REQUIRED"),
+            bigquery.SchemaField("needs_diagnosis_score", "NUMERIC", mode="REQUIRED"),
+            bigquery.SchemaField("platform_score", "NUMERIC", mode="REQUIRED"),
+            bigquery.SchemaField("not_supported_score", "NUMERIC", mode="REQUIRED"),
+            bigquery.SchemaField("total_score", "NUMERIC", mode="REQUIRED"),
+        ]
+
+        self.client.create_table(
+            bigquery.Table(f"{self.client.project}.{history_table}", history_schema),
+            exists_ok=True,
+        )
+
+        query = f"""
+                SELECT recorded_date
+                FROM `{history_table}`
+                ORDER BY recorded_date DESC
+                LIMIT 1
+            """
+
+        rows = list(self.client.query(query).result())
+
+        today = date.today()
+
+        if rows and rows[0]["recorded_date"] >= today:
+            # We've already recorded historic data today
+            return
+
+        query = f"""
+                SELECT *
+                FROM `{metrics_table}`
+            """
+        rows = list(dict(row.items()) for row in self.client.query(query).result())
+        for row in rows:
+            row["recorded_date"] = today
+
+        job_config = bigquery.LoadJobConfig(
```

Comment:
It fails with an error when serializing a date (and decimal) object at `load_table_from_json` with `TypeError: Object of type ... is not JSON serializable`. We can either serialize both, or just use `insert_rows` instead, as it converts objects to strings in the right format internally. 

I've tried this and it works:

```python
logging.info(f"Writing to {history_table} table")

table_ref = self.client.get_table(history_table)
errors = self.client.insert_rows(table_ref, rows)

if errors:
    logging.error(errors)
else:
    logging.info("Metrics history recorded")
    logging.info(f"Loaded {len(rows)} rows into {table_ref}")
```